### ISSUE TITLE
serverSessionManager.asmx shows an error with new Mono 4.2

### ISSUE THREAD

**Issue by tpokorra** [openpetra/openpetra#110]
eg https://demo.solidcharity.com/serverSessionManager.asmx
shows when web.config has `customErrors off`:

````
System.ArgumentNullException
Value cannot be null.
Parameter name: path1

Description: HTTP 500.Error processing request.
Details: Non-web exception. Exception origin (name of application or object): mscorlib.
Exception stack trace:
  at System.IO.Path.Combine (System.String path1, System.String path2) <0x40784ec0 + 0x001c3> in <filename unknown>:0 
  at System.Web.Services.Configuration.WsdlHelpGeneratorElement.get_HelpGeneratorPath () <0x40f908b0 + 0x00034> in <filename unknown>:0 
  at System.Web.Services.Protocols.DocumentationServerProtocol.Initialize () <0x406332e0 + 0x00663> in <filename unknown>:0 
  at System.Web.Services.Protocols.ServerProtocol.SetContext (System.Type type, System.Web.HttpContext context, System.Web.HttpRequest request, System.Web.HttpResponse response) <0x40631ea0 + 0x000fb> in <filename unknown>:0 
  at System.Web.Services.Protocols.ServerProtocolFactory.Create (System.Type type, System.Web.HttpContext context, System.Web.HttpRequest request, System.Web.HttpResponse response, System.Boolean& abortProcessing) <0x40631aa0 + 0x000bb> in <filename unknown>:0 
````

**Comment by tpokorra** [comment]
the code is here: HelpGeneratorPath
https://github.com/mono/referencesource/blob/mono/System.Web.Services/System/Web/Services/Configuration/WsdlHelpGeneratorElement.cs#L39

**Comment by tpokorra** [comment]
see upstream bug: https://bugzilla.xamarin.com/show_bug.cgi?id=37137

**Comment by tpokorra** [comment]
this has been fixed upstream: https://github.com/mono/referencesource/pull/22

**Comment by ColinLaws** [comment]
This bug was just re-introduced. Our development team just attempted to take a web service we previously hosted in IIS and moved to using a Linux server with nginx and fastcgi-mono-server4. On Wednesday March the 16th of 2016, we successfully hosted the web service. To ensure we understood how the service was configured and operating, we wiped our server and started again on Thursday March the 17th of 2016. After wiping the server and re-installing with the latest mono runtime via 'sudo apt-get install mono-complete', the server no longer processed XML based web service requests, returning the exact error above:

-----------------------------------------------
System.ArgumentNullException
Value cannot be null.
Parameter name: path1

Description: HTTP 500.Error processing request.
-----------------------------------------------

Initially under the impression this was our wrong-doing, we were pulling our hair out; however, after finding this bug, wiping our server, and installing our own compiled version of mono 4.3.2.467, the problem went away. We are currently under the assumption that the latest version was accidentally spoiled by a change in the code, or someone merged into your branch with an earlier version.

This bug has been reported at the following:
https://bugzilla.xamarin.com/show_bug.cgi?id=39772
